### PR TITLE
Bug fixes, optimization and miscellanea

### PR DESCRIPTION
**Pull request by kafkaphoenix**
- Added rpgtools to git ignore
- Added spacy requirement to requirements.txt
- Removed rpgtools.py as civsim.py is using the submodule now
- Added ignore=dirty to .gitsubmodule to avoid submodule noise
- Removed unused gitignore options
- Optimized civilisation priorities generation
- Moved in the update method scan method inside if territory check to avoid unnecessary calls
- Added :.2f in the execute method for strings power, technology level and instability to avoid lot of decimals
- make_civ now returns the instance
- Fixed bug: collapse and dissolve methods were creating civilisations in the same spot overwriting previous one (choice(can repeat) > sample (can't repeat))
- Fixed bug: dissolve was creating civilisations and expanding at the same time but sometimes it would crash when calling expand as the civ could not be created (max civ limit)
- Fixed bug: all civ created by dissolve method will expand at the end to avoid one civ expanding over other candidate before being created

### CODE REVIEW

File: civsim.py [hunk 1]
```diff
@@ -114,30 +119,34 @@ def dissolve(self):
...
             CIVLIST.remove(self)
         CIVLETTERS += self.symbol
-        scan(CIV, self.symbol, delete=True)
         if self.territory:
-            for _ in range(random.randint(1, math.ceil(self.powerbase/10)+1)):
-                newseed = random.choice(self.territory)
-                make_civ(newseed[0], newseed[1])
+            scan(CIV, self.symbol, delete=True)
+            n = range(random.randint(1, math.ceil(self.powerbase/10)+1))
+            remanents = random.sample(self.territory, min(len(self.territory), n))
```
Comment (UtilityHotbar) [inline civsim.py]:
```suggestion
            remnant = random.sample(self.territory, min(len(self.territory), n))
```

File: civsim.py [hunk 2]
```diff
@@ -114,30 +119,34 @@ def dissolve(self):
...
+            for remanent in remanents:
```
Comment (UtilityHotbar) [inline civsim.py]:
```suggestion
            for remnant in remnants:
```

File: civsim.py [hunk 3]
```diff
@@ -114,30 +119,34 @@ def dissolve(self):
...
+                make_civ(remanent[0], remanent[1])
```
Comment (UtilityHotbar) [inline civsim.py]:
```suggestion
                make_civ(remnant[0], remnant[1])
```

File: civsim.py [hunk 4]
```diff
@@ -114,30 +119,34 @@ def dissolve(self):
...
+            self.territory.remove(rebel)
+            CIV[rebel[0]][rebel[1]] = UNOCCUPIED
+        if not decimate and rebels:  # Natural disasters do not create new states
+            n = random.randint(1, math.ceil(self.base_power / 10) + 1)
+            warlords = random.sample(rebels, min(len(rebels), n))
+            new_civs = []
+            for warlord in warlords:
+                new_civ = make_civ(warlord[0], warlord[1], parent=self)
+                if new_civ:
+                    new_civs.append(new_civ)
```
Comment (UtilityHotbar) [inline civsim.py]:
```suggestion
        n = random.randint(math.floor(len(self.territory)/1.5), len(self.territory))
        lost_territory = random.sample(self.territory, min(len(self.territory), n))
        for area in lost_territory:
            self.territory.remove(area)
            CIV[area[0]][area[1]] = UNOCCUPIED
        if not decimate and lost_territory:  # Natural disasters do not create new states
            n = random.randint(1, math.ceil(self.base_power / 10) + 1)
            successor_states = random.sample(lost_territory, min(len(lost_territory), n))
            new_civs = []
            for successor_state in successor_states:
                new_civ = make_civ(successor_state[0], successor_state[1], parent=self)
                if new_civ:
                    new_civs.append(new_civ)
```

**Comment by UtilityHotbar** [comment]
I don't think we need to ignore rpgtools in the gitignore. To be more specific, I'm not sure what ignoring would do except make it more confusing when people download the code?

File: .gitignore [hunk 5]
```diff
@@ -1,4 +1,5 @@
 logfile.txt
+rpgtools/
```
Comment (UtilityHotbar) [inline .gitignore]:
```suggestion

```

Comment (kafkaphoenix) [reply .gitignore]: yeah it's not necessary as git knows is a submodule

**Pull request merged**